tests/unit/amo/components/TestGuidesAddonCard.js

```diff
@@ -0,0 +1,248 @@
+import * as React from 'react';
+
+import GuidesAddonCard, {
+  GuidesAddonCardBase,
+} from 'amo/components/GuidesAddonCard';
+import AddonTitle from 'amo/components/AddonTitle';
+import AMInstallButton from 'core/components/AMInstallButton';
+import AddonCompatibilityError from 'amo/components/AddonCompatibilityError';
+import {
+  dispatchClientMetadata,
+  fakeAddon,
+  fakeI18n,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
+import {
+  FATAL_ERROR,
+  INCOMPATIBLE_NOT_FIREFOX,
+  INCOMPATIBLE_UNDER_MIN_VERSION,
+  INSTALLING,
+  UNKNOWN,
+} from 'core/constants';
+import { setInstallError, setInstallState } from 'core/actions/installations';
+import { getErrorMessage } from 'core/utils/addons';
+import { createInternalAddon, loadAddonResults } from 'core/reducers/addons';
+
+describe(__filename, () => {
+  const _loadAddonResults = (addon = createInternalAddon(fakeAddon), store) => {
+    store.dispatch(
+      loadAddonResults({
+        addons: [addon],
+      }),
+    );
+  };
+
+  const getProps = ({
+    addon = createInternalAddon(fakeAddon),
+    addonGuid = addon.guid,
+    addonCustomText = 'Some text',
+    hasAddonManager = true,
+    i18n = fakeI18n(),
+    setCurrentStatus = sinon.spy(),
+    store = dispatchClientMetadata().store,
+    ...customProps
+  } = {}) => {
+    return {
+      addon,
+      addonGuid,
+      addonCustomText,
+      hasAddonManager,
+      i18n,
+      setCurrentStatus,
+      status: UNKNOWN,
+      store,
+      ...customProps,
+    };
+  };
+
+  const render = (customProps = {}) => {
+    const allProps = getProps(customProps);
+    const { addon, store } = allProps;
+
+    if (addon) {
+      _loadAddonResults(addon, store);
```

Comment:
It's confusing to allow `render({ addon })` but then dispatch an action to load it. Dispatching an action is the right thing to do because you need to cover your state mapping logic. You should not render with an `addon` prop because that will bypass the state mapping logic. 

I'd like to see this done explicitly, such as:

```js
// In an example test case:
const addon = { ...fakeAddon };
// Load the add-on into the store.
_loadAddonResults(addon, store);
// Pass a guid just like a real web request would. This will trigger the state mapping code.
const root = render({ addonGuid: addon.guid });
```